src/manager/ResourceManager.js

```diff
@@ -78,30 +81,59 @@ export default class ResourceManager extends AsyncWorker {
         identifier = `${hostname}/${pathParsed.name}-${i}${pathParsed.ext}`;
       }
 
-      if (ok && this.isRelevantAsset(url)) {
+      if (ok) {
         this.lastResourceLoadedAt = Date.now();
-      }
-
-      // collect .js files retrieved for static analysis
-      if (ok && url.endsWith('.js')) {
-        const text = await response.text();
-
-        this.resources.push({
-          ticketId,
-          filename: identifier,
-          data: Buffer.from(text, 'utf8'),
-        });
-      }
-
-      // collect images retrieved for OCR purposes
-      if (ok && isImageUrl(url)) {
-        const buf = await response.buffer();
 
-        this.resources.push({
-          ticketId,
-          filename: identifier,
-          data: Buffer.from(buf),
-        });
+        // collect .js files for display and scanning.
+        if (url.endsWith('.js')) {
+          const text = await response.text();
+
+          this.resources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(text, 'utf8'),
+          });
+
+          this.yaraResources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(text, 'utf8'),
+          });
+
+          // collect images retrieved for OCR purposes
+        } else if (isImageUrl(url)) {
+          const buf = await response.buffer();
+          const text = await response.text();
+
+          this.resources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(buf),
+          });
+
+          // Save images as text for static analysis to detect potential obfuscation
+          this.yaraResources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(text, 'utf8'),
+          });
+          this.yaraResources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(text, 'utf16'),
+          });
+
+          // Process all other resources
+        } else if (request.resourceType() === 'xhr') {
+          const text = await response.text();
+          const charsetMatch = detectCharacterEncoding(text);
+
+          this.yaraResources.push({
+            ticketId,
+            filename: identifier,
+            data: Buffer.from(text, charsetMatch.encoding),
```

Comment:
Could you test this with a binary file (like an image) to see what happens? Wonder how the character encoding detector handles not being able to detect a consistent encoding.

Reply:
sure thing, will get back to you on it.